lib/hacker_news/vote_lookup_service.rb

```diff
@@ -0,0 +1,29 @@
+module HackerNews
+  class VoteLookupService
+
+    attr_reader :user, :votable
+    def initialize(user:, votable: nil, votable_id: nil, votable_type: nil)
+      @user = user
+      @votable = votable || votable_type.to_s.safe_constantize.try(:find_by, id: votable_id)
+    end
+
+    def vote
+      @vote ||= Vote.find_by(user_id: user.id, votable: votable, votable_type: votable.class.to_s)
+    end
+
+    def create_vote
+      if vote.nil? && votable
+        @user.votes.create(votable_type: @votable.class.to_s, votable_id: @votable.id)
```

Comment:
As a best practice, always reference underlying instance variables (`@user` and `@votable`) through reader methods (`user` and `votable`).  There a number of spots in this file you'll want to address.